monitor/launch/launcher.go

```diff
@@ -75,42 +75,52 @@ func (nm *NodeMonitor) GetPid() int {
 	return nm.GetProcess().Pid
 }
 
-// Run starts the node monitor.
-func (nm *NodeMonitor) Run(sockPath, bpfRoot string) {
-	nm.SetTarget(targetName)
-	for {
-		os.Remove(sockPath)
-		if err := syscall.Mkfifo(sockPath, 0600); err != nil {
-			log.WithError(err).Fatalf("Unable to create named pipe %s", sockPath)
-			time.Sleep(time.Duration(5) * time.Second)
-		}
+func (nm *NodeMonitor) run(sockPath, bpfRoot string) error {
+	os.Remove(sockPath)
+	if err := syscall.Mkfifo(sockPath, 0600); err != nil {
+		return fmt.Errorf("Unable to create named pipe %s: %s", sockPath, err)
+	}
+
+	pipe, err := os.OpenFile(sockPath, os.O_RDWR, 0600)
+	if err != nil {
+		return fmt.Errorf("Unable to open named pipe for writing: %s", err)
+	}
+
+	defer pipe.Close()
+
+	nm.Mutex.Lock()
+	nm.pipe = pipe
+	nm.Mutex.Unlock()
+
+	nm.Launcher.SetArgs([]string{"--bpf-root", bpfRoot})
+	nm.Launcher.Run()
 
-		pipe, err := os.OpenFile(sockPath, os.O_RDWR, 0600)
+	r := bufio.NewReader(nm.GetStdout())
+	for nm.GetProcess() != nil {
+		l, err := r.ReadBytes('\n') // this is a blocking read
 		if err != nil {
-			log.WithError(err).Fatal("Unable to open named pipe for writing")
-			time.Sleep(time.Duration(5) * time.Second)
+			return fmt.Errorf("Unable to read stdout from monitor: %s", err)
```

Comment:
I personally don't care, but I've been asked often to hide error messages that are harmless. Perhaps `return nil` with a debug when `err == EOF`.

Reply:
When is EOF harmless?

Reply:
err, you're right. I'm not sure what I was thinking.

Reply:
In that same vein, maybe it's worth adding a log or return an error at the end of this function saying "monitor exited unexpectedly" since the only way to exit this loop without an error is when we notice the monitor not running anymore.